### PR TITLE
Support compile on macos

### PR DESCRIPTION
**Pull request by ulysses-you**
This pr make oap/velox support compile on macos. It seems the changes are not related to vanilla velox.

see gluten pr https://github.com/oap-project/gluten/pull/2691

### CODE REVIEW

File: scripts/setup-helper-functions.sh [hunk 1]
```diff
@@ -141,6 +141,11 @@ function cmake_install {
     -DBUILD_TESTING=OFF \
     "$@"
   ninja -C "${BINARY_DIR}"
-  sudo ninja -C "${BINARY_DIR}" install
+  local OS=`uname -s`
+  if [ $OS == 'Darwin' ]; then
+    ninja -C "${BINARY_DIR}" install
+  else
+    sudo ninja -C "${BINARY_DIR}" install
```
Comment (ulysses-you) [inline scripts/setup-helper-functions.sh:148]: I'm not sure why we need `sudo`. It seems vanilla velox do not have it. I do not know the history so I do not change it in this pr.

File: velox/common/caching/AsyncDataCache.h [hunk 2]
```diff
@@ -36,7 +36,7 @@ namespace facebook::velox::cache {
 class AsyncDataCache;
 class CacheShard;
 class SsdCache;
-class SsdCacheStats;
+struct SsdCacheStats;
```
Comment (ulysses-you) [inline velox/common/caching/AsyncDataCache.h:39]: The Implementation side is `struct`.

File: velox/exec/AggregationHook.h [hunk 3]
```diff
@@ -59,7 +59,7 @@ class AggregationHook : public ValueHook {
 
   std::string toString() const override {
     char buf[256];
-    sprintf(buf, "AggregationHook kind:%d", (int)kind());
+    snprintf(buf, sizeof(buf), "AggregationHook kind:%d", (int)kind());
```
Comment (ulysses-you) [inline velox/exec/AggregationHook.h:62]: `sprintf` is deprecated.

File: velox/type/DecimalUtilOp.h [hunk 4]
```diff
@@ -239,10 +237,7 @@ class DecimalUtilOp {
       if constexpr (std::is_same_v<R, int64_t>) {
         auto result = convertToInt64(result_large, overflow);
         auto remainder = convertToInt64(remainder_large, overflow);
-        if (!(result >= DecimalUtil::kLongDecimalMin &&
-              result <= DecimalUtil::kLongDecimalMax)) {
-          *overflow = true;
-        } else {
+        if (!overflow) {
```
Comment (ulysses-you) [inline velox/type/DecimalUtilOp.h:240]: int64_t should always bigger than the minimum of int128_t and less then the maximum value of int128_t... and it seems the `overflow` has been updated inside `convertToInt64`.

Comment (ulysses-you) [inline velox/type/DecimalUtilOp.h:240]: int64_t should always bigger than the minimum of int128_t and less then the maximum value of int128_t... and it seems the `overflow` has been updated inside `convertToInt64`.

Comment (ulysses-you) [inline velox/type/DecimalUtilOp.h:240]: int64_t should always bigger than the minimum of int128_t and less then the maximum value of int128_t... and it seems the `overflow` has been updated inside `convertToInt64`.

**Comment by ulysses-you** [comment]
```
2:28:55.7150301Z The following tests did not run:
2023-08-10T02:28:55.7150737Z Errors while running CTest
2023-08-10T02:28:55.7151719Z 	 73 - */MemoryPoolTest.memoryLeakCheck/* (Disabled)
2023-08-10T02:28:55.7152517Z 	 74 - */MemoryPoolTest.growBeyondMaxCapacity/* (Disabled)
2023-08-10T02:28:55.7152864Z 
2023-08-10T02:28:55.7153015Z The following tests FAILED:
2023-08-10T02:28:55.7153990Z 	 56 - MemoryManagerTest.addPoolWithArbitrator (Failed)
2023-08-10T02:28:55.7154945Z 	126 - MockSharedArbitrationTest.ensureMemoryPoolMaxCapacity (Subprocess aborted)
2023-08-10T02:28:55.7264264Z 	127 - MockSharedArbitrationTest.ensureNodeMaxCapacity (Subprocess aborted)
2023-08-10T02:28:55.7265338Z 	128 - MockSharedArbitrationTest.failedArbitration (Subprocess aborted)
2023-08-10T02:28:55.7267324Z 	129 - MockSharedArbitrationTest.singlePoolGrowCapacityWithArbitration (Subprocess aborted)
2023-08-10T02:28:55.7268628Z 	132 - MockSharedArbitrationTest.arbitrateBySelfMemoryReclaim (Subprocess aborted)
2023-08-10T02:28:55.7269890Z 	144 - MockSharedArbitrationTest.concurrentArbitrations (Subprocess aborted)
2023-08-10T02:28:55.7271250Z 	145 - MockSharedArbitrationTest.concurrentArbitrationWithTransientRoots (Subprocess aborted)
2023-08-10T02:28:55.7505245Z 	148 - SharedArbitrationTest.reclaimFromCompletedOrderBy (Subprocess aborted)
2023-08-10T02:28:55.7506676Z 	151 - SharedArbitrationTest.reclaimFromCompletedAggregation (Subprocess aborted)
2023-08-10T02:28:55.7508049Z 	154 - SharedArbitrationTest.reclaimFromCompletedJoinBuilder (Subprocess aborted)
2023-08-10T02:28:55.7509158Z 	155 - SharedArbitrationTest.concurrentArbitration (Subprocess aborted)
2023-08-10T02:28:55.7510083Z 	227 - velox_parquet_e2e_filter_test (Failed)
2023-08-10T02:28:55.7510701Z 	242 - velox_functions_test (Failed)
```

The failed test seems irrelevant.

**Comment by Yohahaha** [comment]
MacOS on m1 seems already supported in meta/velox right? have you try that?

**Comment by ulysses-you** [comment]
@Yohahaha yeah, it is. But the code changes in this pr are all from oap/velox...

**Comment by Yohahaha** [comment]
> @Yohahaha yeah, it is. But the code changes in this pr are all from oap/velox...

hmm.. I mean we may wait oap/velox rebase newly meta/velox to support mac build, and these changes seems should contribute to meta/velox. Anyway, I'm not the one who do code rebase work :)

**Comment by ulysses-you** [comment]
also cc @zhouyuan, without this patch the oap/velox does not support compile on macOS (meta/velox supports macOS). I know now gluten does not add more code in oap/velox  repo, but the related code in this pr are all from oap/velox ... that means there is no upstream.

**Comment by zhouyuan** [comment]
@ulysses-you can you please do a rebase? the unit tests should be fixed in main already.

-yuan

**Comment by zhouyuan** [comment]
verified on
https://github.com/oap-project/gluten/pull/2727

**Pull request merged**